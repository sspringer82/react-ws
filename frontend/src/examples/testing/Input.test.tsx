import { fireEvent, render, screen } from '@testing-library/react';
import Input from './Input';
import { expect } from 'vitest';

describe('Input', () => {
  it('should render', () => {
    const { asFragment } = render(<Input />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should change the state', () => {
    render(<Input />);

    fireEvent.change(screen.getByTestId('input'), {
      target: { value: 'Hallo Welt' },
    });

    expect(screen.getByTestId('output')).toHaveTextContent('Hallo Welt');
  });
});
