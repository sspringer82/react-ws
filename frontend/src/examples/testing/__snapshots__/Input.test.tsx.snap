// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > should render 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="output"
    />
    <label>
      Input:
      <input
        data-testid="input"
        value=""
      />
    </label>
  </div>
</DocumentFragment>
`;
